src/main/java/com/iota/iri/network/Node.java

```diff
@@ -103,6 +113,7 @@ public void init() throws Exception {
         parseNeighborsConfig();
 
         executor.submit(spawnBroadcasterThread());
+        executor.submit(spawnRequesterThread());
         executor.submit(spawnTipRequesterThread());
```

Comment:
would it be an idea to rename spawnTipRequesterThread() to spawnLatestMilestoneRequesterThread() for clarity (and adjust the thread name as well)   ?

Reply:
It's not a LatestMilestoneRequester - the tip requester sends the hash of the milestone (request part) along with the data of that milestone (data part).

If we send a transaction that has a hash in its request part that matches the transaction in the data part, the receiving node interprets it as "i don't need anything - please send me a random tip".

So what this thread does is regularly requesting a random tip from its neighbours.

Reply:
aha now i understand this. so it is actually a random tip request and as well sending the latest milestone.

thanks for clearing this up.....i was wondering about this construct in issue #1054  as well.